### PR TITLE
RHCLOUD-27242 Introduce SEDA in webhook connector

### PR DESCRIPTION
**Pull request by gwenneg**
https://camel.apache.org/components/4.0.x/seda-component.html
https://camel.apache.org/components/4.0.x/http-component.html

### CODE REVIEW

File: connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java [hunk 1]
```diff
@@ -10,38 +10,74 @@
...
+    private static final String SEDA_QUEUE_SIZE = "notifications.connector.seda.queue-size";
 
-    @ConfigProperty(name = HTTPS_CONNECT_TIMEOUT_MS, defaultValue = "2500")
-    int httpsConnectTimeout;
+    @ConfigProperty(name = HTTP_CONNECT_TIMEOUT_MS, defaultValue = "2500")
+    int httpConnectTimeout;
 
-    @ConfigProperty(name = HTTPS_SOCKET_TIMEOUT_MS, defaultValue = "2500")
-    int httpsSocketTimeout;
+    @ConfigProperty(name = HTTP_CONNECTIONS_PER_ROUTE, defaultValue = "20")
```
Comment (gwenneg) [inline connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java:24]: Camel's default value.

File: connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java [hunk 2]
```diff
@@ -10,38 +10,74 @@
...
+    int httpConnectionsPerRoute;
+
+    @ConfigProperty(name = HTTP_MAX_TOTAL_CONNECTIONS, defaultValue = "200")
```
Comment (gwenneg) [inline connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java:27]: Camel's default value: https://camel.apache.org/components/4.0.x/http-component.html#_component_option_maxTotalConnections

File: connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java [hunk 3]
```diff
@@ -10,38 +10,74 @@
...
+    int httpMaxTotalConnections;
+
+    @ConfigProperty(name = HTTP_SOCKET_TIMEOUT_MS, defaultValue = "2500")
+    int httpSocketTimeout;
 
     @ConfigProperty(name = ALTERNATIVE_NAMES, defaultValue = "ansible")
     List<String> alternativeNames;
 
+    @ConfigProperty(name = SEDA_CONCURRENT_CONSUMERS, defaultValue = "20")
```
Comment (gwenneg) [inline connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java:36]: NOT Camel's default value: https://camel.apache.org/components/4.0.x/seda-component.html#_component_option_concurrentConsumers

This is purposely aligned with the number of HTTP connections per route.

File: connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java [hunk 4]
```diff
@@ -10,38 +10,74 @@
...
+    int sedaConcurrentConsumers;
+
+    @ConfigProperty(name = SEDA_QUEUE_SIZE, defaultValue = "100")
```
Comment (gwenneg) [inline connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java:39]: NOT Camel's default value: https://camel.apache.org/components/4.0.x/seda-component.html#_component_option_queueSize

File: connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java [hunk 5]
```diff
@@ -10,38 +10,74 @@
...
+    @ConfigProperty(name = HTTP_SOCKET_TIMEOUT_MS, defaultValue = "2500")
+    int httpSocketTimeout;
 
     @ConfigProperty(name = ALTERNATIVE_NAMES, defaultValue = "ansible")
     List<String> alternativeNames;
 
+    @ConfigProperty(name = SEDA_CONCURRENT_CONSUMERS, defaultValue = "20")
+    int sedaConcurrentConsumers;
+
+    @ConfigProperty(name = SEDA_QUEUE_SIZE, defaultValue = "20")
```
Comment (gwenneg) [inline connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookConnectorConfig.java:39]: NOT Camel's default value: https://camel.apache.org/components/4.0.x/seda-component.html#_component_option_queueSize

There's no point accumulating things in memory if we can't process them. The Kafka consumption will simply be paused if the queue is full, just like it is paused today when we're busy performing an HTTP call.

**Review by g-duval — APPROVED** [review]
LGTM

**Pull request merged**